### ISSUE TITLE
Using empty dict as default parameter value

### ISSUE THREAD

**Issue by zpz** [python/cpython#96486]
I just saw `multiprocessing.process.BaseProcess.__init__` has a default arg value `kwargs={}`. I believe this has not caused any issue and probably will never in normal uses of this class. But I think this is a code pattern that we should avoid?

Proposed alternative: `kwargs=None`.

**Comment by stevendaprano** [comment]
We shouldn't be superstitious about programming. There is nothing wrong with the code as it is: the `__init__` method *always* copies or converts the kwargs argument into a new dict: `self._kwargs = dict(kwargs)`. So its fine.

It is not a bug to use a dict as the default value, so this is not a bug. It is not an enhancement. There is no performance benefit. It is just a cosmetic change to the code that would make the code a little bit more complicated, since the method would have to check for None.

The only reason I don't close this ticket immediately is that *technically* there is a very small difference between what [`threading.Thread`](https://github.com/python/cpython/blob/main/Lib/threading.py#L860) and [multiprocessing.process.BaseProcess](https://github.com/python/cpython/blob/main/Lib/multiprocessing/process.py#L71) support. `Thread` *accidentally(?)* supports None as an argument for the kwargs parameters -- it is [not documented](https://docs.python.org/3/library/threading.html#threading.Thread) -- while  `process.BaseProcess` does not.

The two cases are supposed to be "analogous", not necessarily identical. So we have three possibilities:

* Do nothing. Nobody cares about this tiny difference in behaviour. Let it be.
* Decide that the Thread implementation is wrong, and change it to be like `multiprocessing.process.BaseProcess`.
* Decide that the Thread implementation should become official, so we have to change the documentation, write tests for it, and change `multiprocessing.process.BaseProcess` (and other classes?) to match.

I think that the first option (do nothing) is safest, but multiprocessing experts may disagree.

**Comment by zpz** [comment]
The `type-bug` was added by the system; I did not notice it until the issue was created. I overlooked the implementation detail `self._kwargs = dic(kwargs)` but that makes a point: API is more important than implementation.

I do not agree with "it is not a bug to use a dict as the default value". Common pitfalls are pitfalls. It is not a bug "in this case" because of this implementation detail. You did not need to emphasize "it is not...", "it is not ...", "there is no ..." as if you know more and better.

I do not agree with "It is not an enhancement." It depends on the final decision of action. If the decision is to change it, then it is an enhancement.

I do not agree with "Nobody cares...". Why did I care to create this issue? I had better things to do with my time.

I do not agree with option 2. I don't understand why that was even proposed.
